python/cudf/cudf/utils/utils.py

```diff
@@ -396,30 +396,45 @@ def pa_mask_buffer_to_mask(mask_buf, size):
     return Buffer(mask_buf)
 
 
-def pos_from_val(n, start, step, length, side="left"):
-    """Compute the position of a value in an monotonically increasing integral
-    arithmatic series. The value does not need to present in the series. Given
-    parameter ``side``, the value at returned position will either be less or
-    equal, or greater to ``n``.
+def search_range(n, start, step, length, side="left"):
```

Comment:
Suggest the following signature instead:

```python
search_range(start, stop, x, step=1, side="left")
```

For the following reasons:

1. A `range` is expressed in terms of `start`, `stop` and an optional `step`, rather than `start`, `step` and a `length`.

2. Similar functions like `np.searchsorted` and `bisect.bisect` accept the array/list as their _first_ argument and the value to insert as the _second_.

Thoughts?

Reply:
I was also debating which one I should use, my reason picking this signature is for these reasons:

1. It's general to represent a finite arithmetic sequence with `start` `step` and `length`.
2. It's an attempt to inspire other uses of this utility function by conforming to math norm, instead of `range` norm.

Personally I have no preference over the position of `x`.

Reply:
Offline discussion: We think `range` object is more pervasive in cudf as compared to use of APs. Changing function signature now.